python-spec/src/somacore/query/query.py

```diff
@@ -341,8 +381,39 @@ def _read(
             for _xname in all_x_arrays
         }
 
+        def _read_axis_mappings(fn, axis, keys: Sequence[str]) -> Dict[str, np.ndarray]:
+            return {key: fn(axis, key) for key in keys}
+
+        obsm_ft = self._threadpool.submit(
```

Comment:
I'm fine with this solution, but I would submit these tasks before you start the X read, so they all run in parallel.  Otherwise, the read of X and the read of the obsm/... are serialized